apps/portals/e2e/explore.spec.ts

```diff
@@ -109,6 +114,109 @@ const getAvailableFacetsDiv = (page: Page) => {
     .filter({ hasText: 'Available Facets' })
 }
 
+const countVisibleElements = async (allCells: Locator, count: number) => {
+  let visibleCount = 0
+  for (let i = 0; i < count; i++) {
+    const cell = allCells.nth(i)
+    if (await cell.isVisible()) {
+      visibleCount++
+    }
+  }
+  return visibleCount
+}
+
+const toggleChartVisibilityAndVerifyCount = async (page: Page) => {
+  const chartsPanel = page.locator('.FacetNav')
+
+  const allChartCells = chartsPanel.getByRole('figure')
+  const allChartsCount = await allChartCells.count()
+
+  let visibleChartCount = 0
+  let firstVisibleindex = -1
+  for (let i = 0; i < allChartsCount; i++) {
+    const chart = allChartCells.nth(i)
+    if (await chart.isVisible()) {
+      visibleChartCount++
+    }
+    if ((await chart.isVisible()) && firstVisibleindex == -1) {
+      firstVisibleindex = i
+    }
+  }
```

Comment:
Prefer readability here and reuse your function even if it means looping twice; we don't show enough charts to make it worth optimizing to loop only once.

```suggestion
  const visibleChartCount = await countVisibleElements(allChartCells)
  let firstVisibleindex = -1
  for (let i = 0; i < allChartsCount; i++) {
    const chart = allChartCells.nth(i)
    if (await chart.isVisible()) {
      firstVisibleindex = i
      break;
    }
  }
```

Reply:
If `filter` ends up working (comment below), do this:

```suggestion
  const visibleChartCount = await allChartCells.filter({ has: page.locator('visible') }).count()
  const chartToTest = await allChartCells.filter({ has: page.locator('visible') }).first()
  let firstVisibleindex = -1
  for (let i = 0; i < allChartsCount; i++) {
    const chart = allChartCells.nth(i)
    if (await chart.isVisible()) {
      firstVisibleindex = i
      break;
    }
  }
```